scripts/check-commit.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+set -e
+set -x
+echo Testing $(git log -1 --oneline)
+make unit pkg=... case=_NONE_
```

Comment:
This will only compile unit tests (including itests), but not necessarily all non-test code. For example if a package that contains no tests has an error, would we catch that? Perhaps just add a plain `make build` first as well?

Reply:
i'm pretty sure the packages get compiled even if there are no tests to run, i can double check tho!

Reply:
indeed they do get compiled, since the root package imports them even if they have no tests of their own